You are correct about this on `io.h`, it's better to solve all these warnings in future.

Reason about the "strange" and "should really fail" part:


So actually in the windows build log, the issue you proposed has already shown, currently they show as warning, check the [latest windows log](https://github.com/berkeley-abc/abc/actions/runs/27133404442/job/80079735141), there are:
```bash
 D:\a\abc\abc\src\base\main\mainUtils.c(103,11): warning C4013: 'isatty' undefined; assuming extern returning int [D:\a\abc\abc\abclib.vcxproj]
...
D:\a\abc\abc\src\sat\kissat\colors.c(16,35): warning C4013: '_isatty' undefined; assuming extern returning int [D:\a\abc\abc\abclib.vcxproj]
...
D:\a\abc\abc\src\sat\kissat\file.c(37,7): warning C4013: '_access' undefined; assuming extern returning int [D:\a\abc\abc\abclib.vcxproj]
...
D:\a\abc\abc\src\sat\kissat\file.c(60,20): warning C4013: 'S_ISDIR' undefined; assuming extern returning int [D:\a\abc\abc\abclib.vcxproj]
...
``` 
But I assume all of these could be found by [UCRT](https://learn.microsoft.com/en-us/cpp/c-runtime-library/run-time-routines-by-category?view=msvc-170) , which the linker actually search (in the log, we have `C:\Program Files (x86)\Windows Kits\10\include\10.0.26100.0\ucrt`). For `_isatty` and `_access` [here](https://learn.microsoft.com/en-us/cpp/c-runtime-library/file-handling?view=msvc-170) they are handled.
<br>PS: For `S_ISDIR` I'm not sure, it seems no code outside `file.c` calls any function defined in `file.c`.

--- a/.github/scripts/abcexe.vcxproj
+++ b/.github/scripts/abcexe.vcxproj
@@ -33,6 +33,7 @@
     <ClCompile>
       <WarningLevel>Level3</WarningLevel>
       <DisableSpecificWarnings>4146;4334;4996;4703;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <TreatSpecificWarningsAsErrors>4013</TreatSpecificWarningsAsErrors>
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <SDLCheck>true</SDLCheck>

--- a/.github/scripts/abclib.vcxproj.template
+++ b/.github/scripts/abclib.vcxproj.template
@@ -32,6 +32,7 @@
     <ClCompile>
       <WarningLevel>Level3</WarningLevel>
       <DisableSpecificWarnings>4146;4334;4996;4703;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <TreatSpecificWarningsAsErrors>4013</TreatSpecificWarningsAsErrors>
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <SDLCheck>true</SDLCheck>

